src/physics/nonlinear_solid.cpp

```diff
@@ -200,24 +201,45 @@ void NonlinearSolid::advanceTimestep(double& dt)
 
 NonlinearSolid::~NonlinearSolid() {}
 
-void NonlinearSolid::defineInputFileSchema(std::shared_ptr<axom::inlet::SchemaCreator> schema_creator)
+void NonlinearSolid::defineInputFileSchema(axom::inlet::Table& table)
 {
-  auto table = schema_creator->addTable("nonlinear_solid", "Finite deformation solid mechanics module");
+  auto mesh_table = table.addTable("mesh_info", "File location and refinement info for the mesh");
+  serac::mesh::defineInputFileSchema(*mesh_table);
 
   // Polynomial interpolation order
-  table->addInt("order", "Order degree of the finite elements.")->defaultValue(1);
+  table.addInt("order", "Order degree of the finite elements.")->defaultValue(1);
 
   // neo-Hookean material parameters
-  table->addDouble("mu", "Shear modulus in the Neo-Hookean hyperelastic model.")->defaultValue(0.25);
-  table->addDouble("K", "Bulk modulus in the Neo-Hookean hyperelastic model.")->defaultValue(5.0);
+  table.addDouble("mu", "Shear modulus in the Neo-Hookean hyperelastic model.")->defaultValue(0.25);
+  table.addDouble("K", "Bulk modulus in the Neo-Hookean hyperelastic model.")->defaultValue(5.0);
 
   // loading parameters
-  table->addDouble("tx", "Cantilever tip traction in the x direction.")->defaultValue(0.0);
-  table->addDouble("ty", "Cantilever tip traction in the y direction.")->defaultValue(1.0e-3);
-  table->addDouble("tz", "Cantilever tip traction in the z direction.")->defaultValue(0.0);
+  table.addDouble("tx", "Cantilever tip traction in the x direction.")->defaultValue(0.0);
+  table.addDouble("ty", "Cantilever tip traction in the y direction.")->defaultValue(1.0e-3);
+  table.addDouble("tz", "Cantilever tip traction in the z direction.")->defaultValue(0.0);
 
-  auto solver_table = table->addTable("solver", "Linear and Nonlinear Solver Parameters.");
-  serac::EquationSolver::defineInputFileSchema(solver_table);
+  auto solver_table = table.addTable("solver", "Linear and Nonlinear Solver Parameters.");
+  serac::EquationSolver::defineInputFileSchema(*solver_table);
 }
 
 }  // namespace serac
+
+using serac::NonlinearSolid;
+
+NonlinearSolid FromInlet<NonlinearSolid>::operator()(axom::inlet::Table& base)
+{
+  auto mesh = base["mesh_info"].get<std::shared_ptr<mfem::ParMesh>>();
```

Comment:
Once we have multiple physics using the same mesh, this will need to be a shared mesh instance across them all as opposed to each creating a mesh in each of them.

Reply:
Currently we don't inject the member solvers (i.e., `ThermalSolid::ThermalSolid(NonlinearSolid, ThermalCondution)`), so this shouldn't require any changes to the Inlet stuff - we won't be calling `FromInlet<NonlinearSolid>` from `FromInlet<ThermalSolid>`.

Reply:
I think the problem is if you have something like this:
```
main_mesh = {some mesh stuff}

nonlinear_solid = {
    mesh = main_mesh
}

thermal_conduction = {
    mesh = main_mesh 
}
```

This can happen when you have two non-coupled physics modules operating on the same mesh. For example, the solid module could deform the mesh and the thermal solver then operates in a non-coupled fashion on the deformed geometry. This can't happen if we create two separate mesh objects in `FromInlet<NonlinearSolid>` and `FromInlet<ThermalConduction>`.

Reply:
Hmm, we don't have a way of checking referential equality in Inlet but could there ever be a case where the user would set up a problem as you've described and *not* want to have the mesh be shared?  If not, we could "cache" a reference to a constructed mesh object and assume that if a mesh is requested with the same filename, refinement, etc, then the cached mesh should be returned.  Even if that was needed, maybe we could add an optional `unique` flag to the mesh parameters.

Reply:
Probably not. However, this cache option seems a bit convoluted (particularly for a user-facing option).

Reply:
There are going to be complicated objects that just can't be constructed directly from Inlet without jumping through so many hoops that make the program flow and user input convoluted.  This is one of them.

Unless it is a very basic type that can be constructed easily from the template we should be pulling the info out of Inlet as a struct then passing that into the constructor along with the other required information.

In this example, something like this:

```
auto physicsInputInfo = myInlet.get<SomePhysicsInputInfo>(physicsTable);
SomePhysics somePhysics(physicsInputInfo, mesh);
```